classes/ActionScheduler_CanceledAction.php

```diff
@@ -6,10 +6,10 @@
  * Stored action which was canceled and therefore acts like a finished action but should always return a null schedule,
  * regardless of schedule passed to its constructor.
  */
-class ActionScheduler_CanceledAction extends ActionScheduler_FinishedAction {
+class ActionScheduler_CanceledAction extends ActionScheduler_NonexecutableAction {
 
-	public function __construct( $hook, array $args = array(), ActionScheduler_Schedule $schedule = NULL, $group = '', $id = NULL, $status = '', $claim_id = '' ) {
-		parent::__construct( $hook, $args, $schedule, $group );
-		$this->set_schedule( new ActionScheduler_NullSchedule() );
+	public function __construct( $id, $hook, $status, $claim_id = '', array $args = array(), ActionScheduler_Schedule $schedule = NULL, $group = '' ) {
```

Comment:
Sine we're not using the `$schedule` variable at all, it should be removed from the constructor.

Reply:
@JPry this goes back to the old issues originally raised on #88 around having child classes with different constructor param sets. `$schedule` can't be removed without breaking polymorphism of other stored action constructors. e.g. we'd need to update `ActionScheduler_ActionFactory::get_stored_action_instance()` to pass special params for cancelled actions vs. other stored action types. It feels like there should be a neater/cleaner solution to this, but I can't see it yet.